ts/sagas/messages.ts

```diff
@@ -0,0 +1,222 @@
+  } else {
+    // Send and action to store the new message in the store
+    yield put(loadMessageSuccess(response.value));
+    return response.value;
+  }
+}
+
+/**
+ * A generator to load the service details from the Backend
+ *
+ * @param {function} getService - The function that makes the Backend request
+ * @param {string} id - The id of the service to load
+ * @returns {(Error|ServicePublic)}
+ */
+export function* loadService(
+  getService: TypeofApiCall<GetServiceT>,
+  id: string
+): IterableIterator<Effect> {
+  const response: BasicResponseType<ServicePublic> | undefined = yield call(
+    getService,
+    { id }
+  );
+
+  if (!response || response.status !== 200) {
+    return response ? response.value : Error();
+  } else {
+    // Send and action to store the new service in the store
+    yield put(loadServiceSuccess(response.value));
+    return response.value;
+  }
+}
+
+/**
+ * A generator to load messages from the Backend.
+ * The messages returned by the Backend are filtered so the application downloads
+ * only the details of the messages and services not already in the redux store.
+ */
+export function* loadMessages(
+  backendClient: BackendClientT
+): IterableIterator<Effect> {
+  try {
+    // Load already cached messages from the store
+    const cachedMessagesById: MessagesListObject = yield select(
+      messagesByIdSelectors
+    );
+
+    // Load already cached services from the store
+    const cachedServicesById: ServicesListObject = yield select(
+      servicesByIdSelector
+    );
+
+    // Request the list of messages from the Backend
+    const response: BasicResponseType<Messages> | undefined = yield call(
+      backendClient.getMessages,
+      {}
+    );
+
+    /**
+     * If the response is undefined (can't be decoded) or the status is not 200 dispatch a failure action
+     */
+    if (!response || response.status !== 200) {
+      const error: Error = response ? response.value : Error();
+
+      // Dispatch failure action
+      yield put(loadMessagesFailure(error));
+    } else {
+      // Filter messages already in the store
+      const newMessagesWithoutContent = response.value.items.filter(
+        message => !cachedMessagesById.hasOwnProperty(message.id)
+      );
+
+      // Get the messages ids list
+      const newMessagesIds = newMessagesWithoutContent.map(
+        message => message.id
+      );
+
+      // Filter services already in the store
+      const newServicesIds = newMessagesWithoutContent
+        .map(message => message.sender_service_id)
+        .filter(id => !cachedServicesById.hasOwnProperty(id))
+        .filter((id, index, ids) => index === ids.indexOf(id)); // Get unique ids
+
+      // Fetch the services detail in parallel
+      // We fetch services first because to show messages you need the related service info
+      yield all(
+        newServicesIds.map(id =>
+          call(loadService, backendClient.getService, id)
```

Comment:
where does the result of `loadService` go?

Reply:
looks good, just explain it very well since it's not obvious to the reader what's going on :)